### ISSUE TITLE
Support multiple README styles

### ISSUE THREAD

**Issue by j-f1** [j-f1/eslint-docs#10]
TL;DR:

- [x] Load options from the `eslint-config` key in `package.json`
- [x] Add defaults
- [x] Convert the current rule format to use [mustache](https://github.com/janl/mustache.js) for template magic
- Implement options:
  - [x] `includePlugin`
  - [x] `ruleAsCode`
  - [x] `showFixable`
- [x] Implement themes
- [ ] Release!
- [ ] Open PRs as shown below 🚀

Optional:
- allow custom templates
- implement tables ✨

---

Currently, this format is used for the README:

```markdown
* [`{{fullName}}`](./docs/rules/{{name}}.md) — {{description}}
```

However, there are several other formats that could be useful, too:

```markdown
default:
* [{{fullName}}](./docs/rules/{{name}}.md) — {{description}}
name-after:
* {{description}} ([{{fullName}}](./docs/rules/{{name}}.md))
```

Options:
- `includePlugin` should specify whether `{{fullName}}` should be `{{pluginName}}/{{name}}` (`true`) or `{{name}}` (`false`, default)
- `ruleIsCode` should specify whether `{{fullName}}` should be wrapped in `` ` `` (default: `false`).
- `showFixable` should specify whether a piece of text indicating that the rule is fixable should be added to `description` in parentheses. If `false`, it won’t do so. Otherwise, the string value of the option will be inserted.

There should be an option to split up the rules by type:
- [import](https://github.com/benmosher/eslint-plugin-import)
- [Angular](https://github.com/Gillespie59/eslint-plugin-angular)

---

Current usage:

- `default`:
  - [ ] [TypeScript](https://github.com/nzakas/eslint-plugin-typescript) (`includePlugin: true, ruleIsCode: true`)
  - [ ] [unicorn](https://github.com/sindresorhus/eslint-plugin-unicorn) (`fixable: "*fixable*"`)
  - [ ] [ava](https://github.com/avajs/eslint-plugin-ava)
  - [ ] ~[Jest](https://github.com/jest-community/eslint-plugin-jest)~
- `colon`:
  - [ ] [React Native](https://github.com/Intellicode/eslint-plugin-react-native)
  - [ ] [Babel](https://github.com/babel/eslint-plugin-babel): no links (`ruleIsCode: true, showFixable: true, fixable: "🛠"`)
  - [ ] [Lodash](https://github.com/wix/eslint-plugin-lodash)
  - [ ] [ES5](https://github.com/nkt/eslint-plugin-es5): no links (`ruleIsCode: true`)
- `name-after`
  - [ ] [import](https://github.com/benmosher/eslint-plugin-import) (`ruleIsCode: true`)
  - [ ] [Angular](https://github.com/Gillespie59/eslint-plugin-angular)
- Other
  - [ ] [Node](https://github.com/mysticatea/eslint-plugin-node): [custom table script](https://github.com/mysticatea/eslint-plugin-node/blob/master/scripts/update.js)
  - [ ] [ESLint comments](https://github.com/mysticatea/eslint-plugin-eslint-comments): [custom table script](https://github.com/mysticatea/eslint-plugin-eslint-comments/blob/master/scripts/update-readme.js) and [nice additions to the docs template to steal](https://github.com/mysticatea/eslint-plugin-eslint-comments/blob/master/scripts/update-doc-headers.js)
  - [ ] [Ember](https://github.com/ember-cli/eslint-plugin-ember): [custom table script](https://github.com/ember-cli/eslint-plugin-ember/blob/master/scripts/update-rules.js)
  - [ ] [CSS Modules](https://github.com/atfzl/eslint-plugin-css-modules)
  - [ ] [ESLint Plugin](https://github.com/not-an-aardvark/eslint-plugin-eslint-plugin): [custom table script](https://github.com/not-an-aardvark/eslint-plugin-eslint-plugin/blob/master/build/generate-readme-table.js)

Add a rule list:

- [ ] [Ember best practices](https://github.com/ember-best-practices/eslint-plugin-ember-best-practices)
- [ ] [Optimize regex](https://github.com/BrainMaestro/eslint-plugin-optimize-regex)
- [ ] [WordPress](https://github.com/WordPress-Coding-Standards/eslint-plugin-wordpress)
- [ ] [Header](https://github.com/Stuk/eslint-plugin-header)
- [ ] [Prefer Object Spread](https://github.com/bryanrsmith/eslint-plugin-prefer-object-spread)
- [ ] [QUnit](https://github.com/platinumazure/eslint-plugin-qunit)
- [ ] [No Unsanitized](https://github.com/mozilla/eslint-plugin-no-unsanitized)
- [ ] [Expires](https://github.com/digencer/eslint-plugin-expires)
- [ ] [BDD](https://github.com/Nate-Wilkins/eslint-plugin-bdd)
- [ ] [Scissors](https://github.com/MechanicianW/eslint-plugin-scissors)

docs in README; extract to separate files:
- [ ] [SQL](https://github.com/gajus/eslint-plugin-sql)
- [ ] [Ember cleanup](https://github.com/onechiporenko/eslint-plugin-ember-cleanup)
- [ ] [Mocha cleanup](https://github.com/onechiporenko/eslint-plugin-mocha-cleanup)
- [ ] [Reselect](https://github.com/popul/eslint-plugin-reselect)
- [ ] [react-intl](https://github.com/dpvitt/eslint-plugin-reactintl)
- [ ] [variables](https://github.com/frsv/eslint-plugin-variables)
- [ ] [array-func](https://github.com/freaktechnik/eslint-plugin-array-func)

**Comment by j-f1** [comment]
Test

**Comment by j-f1** [comment]
Since I never finished this and since I’ve been a maintainer of Prettier for a while, I think a better solution is to decide on one formatting style and stick to it, although I’ll probably still add an option for showing or hiding categories.

**Comment by armano2** [comment]
@j-f1 i really like this idea, and if you don't mind i'm going to implement this with and options as config, this
